modules/programs/gpg.nix

```diff
@@ -240,44 +258,21 @@ in {
   };
 
   config = mkIf cfg.enable {
-    programs.gpg.settings = {
-      personal-cipher-preferences = mkDefault "AES256 AES192 AES";
-      personal-digest-preferences = mkDefault "SHA512 SHA384 SHA256";
-      personal-compress-preferences = mkDefault "ZLIB BZIP2 ZIP Uncompressed";
-      default-preference-list = mkDefault
-        "SHA512 SHA384 SHA256 AES256 AES192 AES ZLIB BZIP2 ZIP Uncompressed";
-      cert-digest-algo = mkDefault "SHA512";
-      s2k-digest-algo = mkDefault "SHA512";
-      s2k-cipher-algo = mkDefault "AES256";
-      charset = mkDefault "utf-8";
-      fixed-list-mode = mkDefault true;
-      no-comments = mkDefault true;
-      no-emit-version = mkDefault true;
-      keyid-format = mkDefault "0xlong";
-      list-options = mkDefault "show-uid-validity";
-      verify-options = mkDefault "show-uid-validity";
-      with-fingerprint = mkDefault true;
-      require-cross-certification = mkDefault true;
-      no-symkey-cache = mkDefault true;
-      use-agent = mkDefault true;
-    };
```

Comment:
This change doesn't look backwards compatible... Are these the defaults that GPG uses anyway?

Reply:
It's absulutely not the defaults that GPG uses, it will cause https://github.com/nix-community/home-manager/issues/5383 , and GPG's defaults not. So it should be deleted whatever the backwards compatible.

Reply:
ok, I unchecked the backwards compatible checkbox in the PR description. @rycee do we need to gate this behind a `stateVersion`?